source/poggles/texture.cpp

```diff
@@ -40,86 +13,100 @@ struct overload : Ts...
 template<class... Ts>
 overload(Ts...) -> overload<Ts...>;
 
-poggles::texture::texture(GLenum target)
-    : m_original_target(target)
+auto poggles::loadFromFile(poggles::texture& tex,
+                           std::filesystem::path const& filename,
+                           std::optional<GLenum> target,
+                           GLint level) -> bool
 {
-  bind();
+  int width = 0;
+  int height = 0;
+  int channels = 0;
+  uint8_t* data = stbi_load(filename.c_str(), &width, &height, &channels, 0);
+  if (data != nullptr) {
+    if (target.has_value()) {
+      tex.target = *target;
+    }
+    // Call ourselves, but with the relevant data extracted from the
+    // file
+    bool status =
+        loadFromData(tex, data, width, height, channels, target, level);
+    stbi_image_free(data);
+    return status;
+  } else {
+    return false;
+  }
+}
+
+auto poggles::loadFromData(poggles::texture& tex,
+                           std::variant<float*, uint8_t*> data,
+                           int width,
+                           int height,
+                           int channels,
+                           std::optional<GLenum> target,
+                           GLint level) -> bool
+{
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
```

Comment:
Should this be `target` (if target is non-empty?)

Reply:
Good catch, it's fixed now